packages/snaps-controllers/src/interface/SnapInterfaceController.ts

```diff
@@ -132,7 +140,22 @@ export class SnapInterfaceController extends BaseController<
     super({
       messenger,
       metadata: {
-        interfaces: { persist: false, anonymous: false },
+        interfaces: {
+          persist: (interfaces: Record<string, StoredInterface>) => {
+            return Object.entries(interfaces).reduce<
+              Record<string, StoredInterface>
+            >((persistedInterfaces, [id, snapInterface]) => {
+              switch (snapInterface.contentType) {
+                case ContentType.Notification:
```

Comment:
How are notification interfaces ever deleted?

Reply:
Interface cleanup has to be done in the extension (at the end of the notification timeout). We don't have access to the controller messenger in the constructor so can't grab the notifications themselves and see what's expired from here.

Reply:
Hmm, for other interfaces we do this in the extension but that is because we have no other choice. It may make sense to listen for an event or something similar from the NotificationController to delete the persisted interfaces instead? 🤔

Reply:
Oooo I like that, yeah I can do that.

Reply:
I think that should come in a different PR, I don't want to add in logic for the `NotificationController` when it's on the way out. That controller doesn't emit events either.

Reply:
I was referring to the `NotificationServicesController` which we will be using from now on.

Reply:
[e005581](https://github.com/MetaMask/snaps/pull/2856/commits/e00558172cb870ba252808dd63bf616af9e9f075)